AngouriMath/Core/Entity.cs

```diff
@@ -521,5 +553,24 @@ public Tensor Inverse()
                 func(New(Expression.Replace(func), Var.Replace(func), Destination.Replace(func), ApproachFrom));
             protected override Entity[] InitDirectChildren() => new[] { Expression, Var, Destination };
         }
+        #endregion
+
+        #region Discrete
+
+        public partial record Signumf(Entity Argument) : Function
+        {
+            public override Entity Replace(Func<Entity, Entity> func) => func(Argument.Replace(func).Signum());
+            protected override Entity[] InitDirectChildren() => new[] { Argument };
+        }
+        public Entity Signum() => new Signumf(this);
+
+        public partial record Absf(Entity Argument) : Function
+        {
+            public override Entity Replace(Func<Entity, Entity> func) => func(Argument.Replace(func).Abs());
+            protected override Entity[] InitDirectChildren() => new[] { Argument };
+        }
+        public Entity Abs() => new Absf(this);
+
```

Comment:
Why didn't you copy the `New` above? As `ReferenceEquals` is used, if a new object is returned, the cache will be thrown away.

Reply:
I guess it should be forced via abstract methods. Ill czech it